HPCC-23825 Ensure each action activity is unique

ghalliday opened the pull request:
Signed-off-by: Gavin Halliday <gavin.halliday@lexisnexis.com>

<!-- Thank you for submitting a pull request to the HPCC project

 PLEASE READ the following before proceeding.

 This project only accepts pull requests related to open JIRA issues.
 If suggesting a new feature or change, please discuss it in a JIRA issue first.
 If fixing a bug, there should be an issue describing it with steps to reproduce.
 The title line of the pull request (and of each commit within it) should refer to the
 associated issue using the format:

 HPCC-nnnnn Short description of issue

 This will allow the Jira ticket to be automatically updated to refer to this pull request,
 and will ensure that the automatically-generated changelog is properly formatted.
 Where a pull request contains a single commit the pull request title will be set automatically,
 assuming that the commit has followed the proper guidelines.

 Please go over all the following points, and put an `x` in all the boxes that apply. You may find
 it easier to press the 'Create' button first then click on the checkboxes to edit the comment.
-->

## Type of change:
- [ ] This change is a bug fix (non-breaking change which fixes an issue).
- [ ] This change is a new feature (non-breaking change which adds functionality).
- [x] This change improves the code (refactor or other change that does not change the functionality)
- [ ] This change fixes warnings (the fix does not alter the functionality or the generated code)
- [ ] This change is a breaking change (fix or feature that will cause existing behavior to change).
- [ ] This change alters the query API (existing queries will have to be recompiled)

## Checklist:
- [x] My code follows the code style of this project.
  - [x] My code does not create any new warnings from compiler, build system, or lint.
- [x] The commit message is properly formatted and free of typos.
  - [x] The commit message title makes sense in a changelog, by itself.
  - [x] The commit is signed.
- [ ] My change requires a change to the documentation.
  - [ ] I have updated the documentation accordingly, or...
  - [ ] I have created a JIRA ticket to update the documentation.
  - [ ] Any new interfaces or exported functions are appropriately commented.
- [x] I have read the CONTRIBUTORS document.
- [ ] The change has been fully tested:
  - [ ] I have added tests to cover my changes.
  - [ ] All new and existing tests passed.
  - [ ] I have checked that this change does not introduce memory leaks.
  - [ ] I have used Valgrind or similar tools to check for potential issues.
- [ ] I have given due consideration to all of the following potential concerns:
  - [ ] Scalability
  - [ ] Performance
  - [ ] Security
  - [ ] Thread-safety
  - [ ] Cloud-compatibility
  - [ ] Premature optimization
  - [ ] Existing deployed queries will not be broken
  - [ ] This change fixes the problem, not just the symptom
  - [ ] The target branch of this pull request is appropriate for such a change.
- [ ] There are no similar instances of the same problem that should be addressed
  - [ ] I have addressed them here
  - [ ] I have raised JIRA issues to address them separately
- [ ] This is a user interface / front-end modification
  - [ ] I have tested my changes in multiple modern browsers
  - [ ] The component(s) render as expected

## Smoketest:
- [ ] Send notifications about my Pull Request position in Smoketest queue.
- [ ] Test my draft Pull Request.

## Testing:
<!-- Please describe how this change has been tested.-->

<!-- Thank you for taking the time to submit this pull request and to answer all of the above-->

ghalliday committed:
HPCC-23825 Ensure each action activity is unique

Signed-off-by: Gavin Halliday <gavin.halliday@lexisnexis.com>

ghalliday commented:
@shamser please review.  Likely to need some discussion to understand the changes.

HPCCSmoketest commented:
Process of PR-13589, label: uniqueaction starts now.
The reason of this test is: Forced to re-test.
Commit ID: fdd6bec438e481b286b1a3ce3bdac61db2858a87
Estimated completion time is ~0.5 hour(s)
centos 7.6.1810 (Linux 3.10.0-957.1.3.el7.x86_64)
 GCC: gcc (GCC) 7.3.1 20180303 (Red Hat 7.3.1-5)
 Host: ip-10-20-0-190.ca-central-1.compute.internal

HPCCSmoketest commented:
Automated Smoketest: :white_check_mark:
OS: centos 7.6.1810 (Linux 3.10.0-957.1.3.el7.x86_64)
 GCC: gcc (GCC) 7.3.1 20180303 (Red Hat 7.3.1-5)
 Host: ip-10-20-0-190.ca-central-1.compute.internal
Sha: fdd6bec438e481b286b1a3ce3bdac61db2858a87
Build: success
Milestone:Install hpccsystems-platform-community_7.9.0-trunk0.el7.x86_64.rpm
HPCC Start: OK

Unit tests result:

| Test | total | passed | failed | errors | timeout | elaps | 
|---|---|---|---|---|---|---|
|unittest |120 |120 |0 |0 |0 |53 sec |
|wutoolTest(Dali) |19 |19 |0 |0 |0 |1 sec |
|wutoolTest(Cassandra) |19 |19 |0 |0 |0 |8 sec |



Regression test result:

| phase | total | pass | fail | elaps | 
|---|---|---|---|---|
|setup (hthor) |11 |11 |0 |24 sec  (00:00:24) |
|setup (thor) |11 |11 |0 |42 sec  (00:00:42) |
|setup (roxie) |11 |11 |0 |16 sec  (00:00:16) |
|test (hthor) |873 |873 |0 |379 sec  (00:06:19) |
|test (thor) |812 |812 |0 |743 sec  (00:12:23) |
|test (roxie) |953 |953 |0 |389 sec  (00:06:29) |


HPCC Stop: OK
HPCC Uninstall: OK
Time stats:

| Prep time | Build time | Package time | Install time | Start time | Test time | Stop time | Summary | 
|---|---|---|---|---|---|---|---|
|   10 sec (00:00:10) |  643 sec (00:10:43) |  117 sec (00:01:57) |   23 sec (00:00:23) |   20 sec (00:00:20) | 1830 sec (00:30:30) |   19 sec (00:00:19) | 2662 sec (00:44:22) |

HPCCSmoketest commented:
Process of PR-13589, label: uniqueaction starts now.
The reason of this test is: Did not build yet.
Commit ID: fdd6bec438e481b286b1a3ce3bdac61db2858a87
Estimated completion time is ~0.9 hour(s)
centos 7.7.1908 (Linux 3.10.0-957.1.3.el7.x86_64)
 GCC: gcc (GCC) 7.3.1 20180303 (Red Hat 7.3.1-5)
 Host: ip-10-20-0-116.ca-central-1.compute.internal

HPCCSmoketest commented:
Automated Smoketest: :white_check_mark:
OS: centos 7.7.1908 (Linux 3.10.0-957.1.3.el7.x86_64)
 GCC: gcc (GCC) 7.3.1 20180303 (Red Hat 7.3.1-5)
 Host: ip-10-20-0-116.ca-central-1.compute.internal
Sha: fdd6bec438e481b286b1a3ce3bdac61db2858a87
Build: success
Milestone:Install hpccsystems-platform-community_7.9.0-trunk0.el7.x86_64.rpm
HPCC Start: OK

Unit tests result:

| Test | total | passed | failed | errors | timeout | elaps | 
|---|---|---|---|---|---|---|
|unittest |120 |120 |0 |0 |0 |53 sec |
|wutoolTest(Dali) |19 |19 |0 |0 |0 |2 sec |
|wutoolTest(Cassandra) |19 |19 |0 |0 |0 |9 sec |



Regression test result:

| phase | total | pass | fail | elaps | 
|---|---|---|---|---|
|setup (hthor) |11 |11 |0 |28 sec  (00:00:28) |
|setup (thor) |11 |11 |0 |44 sec  (00:00:44) |
|setup (roxie) |11 |11 |0 |20 sec  (00:00:20) |
|test (hthor) |873 |873 |0 |580 sec  (00:09:40) |
|test (thor) |812 |812 |0 |1009 sec  (00:16:49) |
|test (roxie) |953 |953 |0 |729 sec  (00:12:09) |


HPCC Stop: OK
HPCC Uninstall: OK
Time stats:

| Prep time | Build time | Package time | Install time | Start time | Test time | Stop time | Summary | 
|---|---|---|---|---|---|---|---|
|    9 sec (00:00:09) |  937 sec (00:15:37) |  122 sec (00:02:02) |   23 sec (00:00:23) |   17 sec (00:00:17) | 2649 sec (00:44:09) |   18 sec (00:00:18) | 3775 sec (01:02:55) |

shamser reviewed (approved):
@richardkchapman Reviewed.

HPCCSmoketest commented:
Process of PR-13589, label: uniqueaction starts now.
The reason of this test is: Forced to re-test.
Commit ID: fdd6bec438e481b286b1a3ce3bdac61db2858a87
Estimated completion time is ~0.8 hour(s)
centos 7.6.1810 (Linux 3.10.0-957.1.3.el7.x86_64)
 GCC: gcc (GCC) 7.3.1 20180303 (Red Hat 7.3.1-5)
 Host: ip-10-20-0-218.ca-central-1.compute.internal

HPCCSmoketest commented:
Automated Smoketest: :x:
OS: centos 7.6.1810 (Linux 3.10.0-957.1.3.el7.x86_64)
 GCC: gcc (GCC) 7.3.1 20180303 (Red Hat 7.3.1-5)
 Host: ip-10-20-0-218.ca-central-1.compute.internal
Sha: fdd6bec438e481b286b1a3ce3bdac61db2858a87
Build: success
Milestone:Install hpccsystems-platform-community_7.9.0-trunk0.el7.x86_64.rpm
HPCC Start: OK

Unit tests result:

| Test | total | passed | failed | errors | timeout | elaps | 
|---|---|---|---|---|---|---|
|unittest |136 |136 |0 |0 |0 |72 sec |
|wutoolTest(Dali) |19 |19 |0 |0 |0 |2 sec |
|wutoolTest(Cassandra) |19 |19 |0 |0 |0 |7 sec |



Regression test result:

| phase | total | pass | fail | elaps | 
|---|---|---|---|---|
|setup (hthor) |11 |11 |0 |24 sec  (00:00:24) |
|setup (thor) |11 |11 |0 |45 sec  (00:00:45) |
|setup (roxie) |11 |11 |0 |16 sec  (00:00:16) |
|test (hthor) |882 |881 |1 |335 sec  (00:05:35) |
|test (thor) |820 |819 |1 |858 sec  (00:14:18) |
|test (roxie) |961 |960 |1 |407 sec  (00:06:47) |


[Errors:](https://gist.github.com/3544255873be88e74604feae5972d6d4)
- test
  - hthor
    - 863\. [workflow_5](https://gist.github.com/HPCCSmoketest/3544255873be88e74604feae5972d6d4#file-workflow_5-w20200617-002958-log), 
  - thor
    - 802\. [workflow_5](https://gist.github.com/HPCCSmoketest/3544255873be88e74604feae5972d6d4#file-workflow_5-w20200617-004433-log), 
  - roxie
    - 944\. [workflow_5](https://gist.github.com/HPCCSmoketest/3544255873be88e74604feae5972d6d4#file-workflow_5-w20200617-005116-4-log), 

HPCC Stop: OK
HPCC Uninstall: OK
Time stats:

| Prep time | Build time | Package time | Install time | Start time | Test time | Stop time | Summary | 
|---|---|---|---|---|---|---|---|
|    9 sec (00:00:09) |  732 sec (00:12:12) |  118 sec (00:01:58) |   23 sec (00:00:23) |   18 sec (00:00:18) | 1940 sec (00:32:20) |   20 sec (00:00:20) | 2860 sec (00:47:40) |

HPCCSmoketest commented:
Process of PR-13589, label: uniqueaction starts now.
The reason of this test is: Forced to re-test.
Commit ID: fdd6bec438e481b286b1a3ce3bdac61db2858a87
Estimated completion time is ~0.8 hour(s)
centos 7.6.1810 (Linux 3.10.0-957.1.3.el7.x86_64)
 GCC: gcc (GCC) 7.3.1 20180303 (Red Hat 7.3.1-5)
 Host: ip-10-20-0-194.ca-central-1.compute.internal

HPCCSmoketest commented:
Automated Smoketest: :x:
OS: centos 7.6.1810 (Linux 3.10.0-957.1.3.el7.x86_64)
 GCC: gcc (GCC) 7.3.1 20180303 (Red Hat 7.3.1-5)
 Host: ip-10-20-0-194.ca-central-1.compute.internal
Sha: fdd6bec438e481b286b1a3ce3bdac61db2858a87
Build: success
Milestone:Install hpccsystems-platform-community_7.9.0-trunk0.el7.x86_64.rpm
HPCC Start: OK

Unit tests result:

| Test | total | passed | failed | errors | timeout | elaps | 
|---|---|---|---|---|---|---|
|unittest |136 |136 |0 |0 |0 |75 sec |
|wutoolTest(Dali) |19 |19 |0 |0 |0 |2 sec |
|wutoolTest(Cassandra) |19 |19 |0 |0 |0 |7 sec |



Regression test result:

| phase | total | pass | fail | elaps | 
|---|---|---|---|---|
|setup (hthor) |11 |11 |0 |25 sec  (00:00:25) |
|setup (thor) |11 |11 |0 |45 sec  (00:00:45) |
|setup (roxie) |11 |11 |0 |16 sec  (00:00:16) |
|test (hthor) |882 |881 |1 |330 sec  (00:05:30) |
|test (thor) |820 |819 |1 |849 sec  (00:14:09) |
|test (roxie) |961 |960 |1 |424 sec  (00:07:04) |


[Errors:](https://gist.github.com/ddbac5fc6c64c7daebec9ac54a234342)
- test
  - hthor
    - 863\. [workflow_5](https://gist.github.com/HPCCSmoketest/ddbac5fc6c64c7daebec9ac54a234342#file-workflow_5-w20200617-012406-2-log), 
  - thor
    - 802\. [workflow_5](https://gist.github.com/HPCCSmoketest/ddbac5fc6c64c7daebec9ac54a234342#file-workflow_5-w20200617-013833-2-log), 
  - roxie
    - 944\. [workflow_5](https://gist.github.com/HPCCSmoketest/ddbac5fc6c64c7daebec9ac54a234342#file-workflow_5-w20200617-014509-4-log), 

HPCC Stop: OK
HPCC Uninstall: OK
Time stats:

| Prep time | Build time | Package time | Install time | Start time | Test time | Stop time | Summary | 
|---|---|---|---|---|---|---|---|
|    9 sec (00:00:09) |  726 sec (00:12:06) |  118 sec (00:01:58) |   22 sec (00:00:22) |   17 sec (00:00:17) | 1948 sec (00:32:28) |   18 sec (00:00:18) | 2858 sec (00:47:38) |

richardkchapman commented:
@ghalliday This appears to be failing some tests...

HPCCSmoketest commented:
Process of PR-13589, label: uniqueaction starts now.
The reason of this test is: New commit.
Commit ID: 466e33cdd80d65e09de22dd69dc2fd0d4c56f31c
Estimated completion time is ~0.9 hour(s)
centos 7.8.2003 (Linux 3.10.0-957.1.3.el7.x86_64)
 GCC: gcc (GCC) 7.3.1 20180303 (Red Hat 7.3.1-5)
 Host: ip-10-20-0-116.ca-central-1.compute.internal

HPCCSmoketest commented:
Automated Smoketest: :x:
OS: centos 7.8.2003 (Linux 3.10.0-957.1.3.el7.x86_64)
 GCC: gcc (GCC) 7.3.1 20180303 (Red Hat 7.3.1-5)
 Host: ip-10-20-0-116.ca-central-1.compute.internal
Sha: 466e33cdd80d65e09de22dd69dc2fd0d4c56f31c
Build: success
Milestone:Install hpccsystems-platform-community_7.13.0-trunk0.el7.x86_64.rpm
HPCC Start: OK

Unit tests result:

| Test | total | passed | failed | errors | timeout | elaps | 
|---|---|---|---|---|---|---|
|unittest |145 |145 |0 |0 |0 |74 sec |
|wutoolTest(Dali) |19 |19 |0 |0 |0 |1 sec |
|wutoolTest(Cassandra) |19 |19 |0 |0 |0 |9 sec |



Regression test result:

| phase | total | pass | fail | elaps | 
|---|---|---|---|---|
|setup (hthor) |11 |11 |0 |29 sec  (00:00:29) |
|setup (thor) |11 |11 |0 |45 sec  (00:00:45) |
|setup (roxie) |11 |11 |0 |19 sec  (00:00:19) |
|test (hthor) |934 |934 |0 |671 sec  (00:11:11) |
|test (thor) |845 |845 |0 |1195 sec  (00:19:55) |
|test (roxie) |1013 |1009 |4 |772 sec  (00:12:52) |


[Errors:](https://gist.github.com/82a9a5ce2cddbe02f6b38b32dad440ec)
- test
  - roxie
    - 10\. [action1](https://gist.github.com/HPCCSmoketest/82a9a5ce2cddbe02f6b38b32dad440ec#file-action1-w20200928-164615-8-log), 11\. [action1a](https://gist.github.com/HPCCSmoketest/82a9a5ce2cddbe02f6b38b32dad440ec#file-action1a-w20200928-164617-log), 13\. [action4](https://gist.github.com/HPCCSmoketest/82a9a5ce2cddbe02f6b38b32dad440ec#file-action4-w20200928-164618-log), 14\. [action5](https://gist.github.com/HPCCSmoketest/82a9a5ce2cddbe02f6b38b32dad440ec#file-action5-w20200928-164618-7-log), 

HPCC Stop: OK
HPCC Uninstall: OK
Time stats:

| Prep time | Build time | Package time | Install time | Start time | Test time | Stop time | Summary | 
|---|---|---|---|---|---|---|---|
|    9 sec (00:00:09) | 1071 sec (00:17:51) |  126 sec (00:02:06) |   24 sec (00:00:24) |   18 sec (00:00:18) | 2986 sec (00:49:46) |   18 sec (00:00:18) | 4252 sec (01:10:52) |

HPCCSmoketest commented:
Process of PR-13589, label: uniqueaction starts now.
The reason of this test is: Forced to re-test.
Commit ID: 466e33cdd80d65e09de22dd69dc2fd0d4c56f31c
Estimated completion time is ~0.8 hour(s)
centos 7.6.1810 (Linux 3.10.0-957.1.3.el7.x86_64)
 GCC: gcc (GCC) 7.3.1 20180303 (Red Hat 7.3.1-5)
 Host: ip-10-20-0-117.ca-central-1.compute.internal

HPCCSmoketest commented:
Automated Smoketest: :x:
OS: centos 7.6.1810 (Linux 3.10.0-957.1.3.el7.x86_64)
 GCC: gcc (GCC) 7.3.1 20180303 (Red Hat 7.3.1-5)
 Host: ip-10-20-0-117.ca-central-1.compute.internal
Sha: 466e33cdd80d65e09de22dd69dc2fd0d4c56f31c
Build: success
Milestone:Install hpccsystems-platform-community_7.13.0-trunk0.el7.x86_64.rpm
HPCC Start: OK

Unit tests result:

| Test | total | passed | failed | errors | timeout | elaps | 
|---|---|---|---|---|---|---|
|unittest |145 |145 |0 |0 |0 |75 sec |
|wutoolTest(Dali) |19 |19 |0 |0 |0 |2 sec |
|wutoolTest(Cassandra) |19 |19 |0 |0 |0 |7 sec |



Regression test result:

| phase | total | pass | fail | elaps | 
|---|---|---|---|---|
|setup (hthor) |11 |11 |0 |28 sec  (00:00:28) |
|setup (thor) |11 |11 |0 |44 sec  (00:00:44) |
|setup (roxie) |11 |11 |0 |19 sec  (00:00:19) |
|test (hthor) |934 |934 |0 |382 sec  (00:06:22) |
|test (thor) |845 |845 |0 |895 sec  (00:14:55) |
|test (roxie) |1013 |1009 |4 |442 sec  (00:07:22) |


[Errors:](https://gist.github.com/3d2b0333949725a93370cde61383c2b7)
- test
  - roxie
    - 10\. [action1](https://gist.github.com/HPCCSmoketest/3d2b0333949725a93370cde61383c2b7#file-action1-w20200928-195613-1-log), 11\. [action1a](https://gist.github.com/HPCCSmoketest/3d2b0333949725a93370cde61383c2b7#file-action1a-w20200928-195615-log), 13\. [action4](https://gist.github.com/HPCCSmoketest/3d2b0333949725a93370cde61383c2b7#file-action4-w20200928-195615-3-log), 14\. [action5](https://gist.github.com/HPCCSmoketest/3d2b0333949725a93370cde61383c2b7#file-action5-w20200928-195615-2-log), 

HPCC Stop: OK
HPCC Uninstall: OK
Time stats:

| Prep time | Build time | Package time | Install time | Start time | Test time | Stop time | Summary | 
|---|---|---|---|---|---|---|---|
|    9 sec (00:00:09) |  770 sec (00:12:50) |  121 sec (00:02:01) |   23 sec (00:00:23) |   18 sec (00:00:18) | 2066 sec (00:34:26) |   19 sec (00:00:19) | 3026 sec (00:50:26) |

HPCCSmoketest commented:
Process of PR-13589, label: uniqueaction starts now.
The reason of this test is: Forced to re-test.
Commit ID: f664df19163b4a3aa1af9a5d91a984605b88c300
Estimated completion time is ~0.8 hour(s)
centos 7.6.1810 (Linux 3.10.0-957.1.3.el7.x86_64)
 GCC: gcc (GCC) 7.3.1 20180303 (Red Hat 7.3.1-5)
 Host: ip-10-20-0-221.ca-central-1.compute.internal

HPCCSmoketest commented:
Process of PR-13589, label: uniqueaction starts now.
The reason of this test is: New commit.
Commit ID: f664df19163b4a3aa1af9a5d91a984605b88c300
Estimated completion time is ~0.9 hour(s)
centos 7.8.2003 (Linux 3.10.0-957.1.3.el7.x86_64)
 GCC: gcc (GCC) 7.3.1 20180303 (Red Hat 7.3.1-5)
 Host: ip-10-20-0-116.ca-central-1.compute.internal

HPCCSmoketest commented:
Automated Smoketest: :white_check_mark:
OS: centos 7.6.1810 (Linux 3.10.0-957.1.3.el7.x86_64)
 GCC: gcc (GCC) 7.3.1 20180303 (Red Hat 7.3.1-5)
 Host: ip-10-20-0-221.ca-central-1.compute.internal
Sha: f664df19163b4a3aa1af9a5d91a984605b88c300
Build: success
Milestone:Install hpccsystems-platform-community_7.13.0-trunk0.el7.x86_64.rpm
HPCC Start: OK

Unit tests result:

| Test | total | passed | failed | errors | timeout | elaps | 
|---|---|---|---|---|---|---|
|unittest |145 |145 |0 |0 |0 |73 sec |
|wutoolTest(Dali) |19 |19 |0 |0 |0 |2 sec |
|wutoolTest(Cassandra) |19 |19 |0 |0 |0 |7 sec |



Regression test result:

| phase | total | pass | fail | elaps | 
|---|---|---|---|---|
|setup (hthor) |11 |11 |0 |29 sec  (00:00:29) |
|setup (thor) |11 |11 |0 |43 sec  (00:00:43) |
|setup (roxie) |11 |11 |0 |18 sec  (00:00:18) |
|test (hthor) |934 |934 |0 |383 sec  (00:06:23) |
|test (thor) |845 |845 |0 |894 sec  (00:14:54) |
|test (roxie) |1013 |1013 |0 |429 sec  (00:07:09) |


HPCC Stop: OK
HPCC Uninstall: OK
Time stats:

| Prep time | Build time | Package time | Install time | Start time | Test time | Stop time | Summary | 
|---|---|---|---|---|---|---|---|
|    9 sec (00:00:09) |  773 sec (00:12:53) |  130 sec (00:02:10) |   23 sec (00:00:23) |   19 sec (00:00:19) | 2040 sec (00:34:00) |   19 sec (00:00:19) | 3013 sec (00:50:13) |

HPCCSmoketest commented:
Automated Smoketest: :white_check_mark:
OS: centos 7.8.2003 (Linux 3.10.0-957.1.3.el7.x86_64)
 GCC: gcc (GCC) 7.3.1 20180303 (Red Hat 7.3.1-5)
 Host: ip-10-20-0-116.ca-central-1.compute.internal
Sha: f664df19163b4a3aa1af9a5d91a984605b88c300
Build: success
Milestone:Install hpccsystems-platform-community_7.13.0-trunk0.el7.x86_64.rpm
HPCC Start: OK

Unit tests result:

| Test | total | passed | failed | errors | timeout | elaps | 
|---|---|---|---|---|---|---|
|unittest |145 |145 |0 |0 |0 |74 sec |
|wutoolTest(Dali) |19 |19 |0 |0 |0 |2 sec |
|wutoolTest(Cassandra) |19 |19 |0 |0 |0 |8 sec |



Regression test result:

| phase | total | pass | fail | elaps | 
|---|---|---|---|---|
|setup (hthor) |11 |11 |0 |29 sec  (00:00:29) |
|setup (thor) |11 |11 |0 |44 sec  (00:00:44) |
|setup (roxie) |11 |11 |0 |19 sec  (00:00:19) |
|test (hthor) |934 |934 |0 |662 sec  (00:11:02) |
|test (thor) |845 |845 |0 |1185 sec  (00:19:45) |
|test (roxie) |1013 |1013 |0 |781 sec  (00:13:01) |


HPCC Stop: OK
HPCC Uninstall: OK
Time stats:

| Prep time | Build time | Package time | Install time | Start time | Test time | Stop time | Summary | 
|---|---|---|---|---|---|---|---|
|    9 sec (00:00:09) | 1087 sec (00:18:07) |  127 sec (00:02:07) |   24 sec (00:00:24) |   18 sec (00:00:18) | 2997 sec (00:49:57) |   17 sec (00:00:17) | 4279 sec (01:11:19) |

HPCCSmoketest commented:
Process of PR-13589, label: uniqueaction starts now.
The reason of this test is: Forced to re-test.
Commit ID: f725998e723b13e5e20093ebe3a7dd7146dca99f
Estimated completion time is ~0.8 hour(s)
centos 7.6.1810 (Linux 3.10.0-957.1.3.el7.x86_64)
 GCC: gcc (GCC) 7.3.1 20180303 (Red Hat 7.3.1-5)
 Host: ip-10-20-0-65.ca-central-1.compute.internal

HPCCSmoketest commented:
Process of PR-13589, label: uniqueaction starts now.
The reason of this test is: New commit.
Commit ID: f725998e723b13e5e20093ebe3a7dd7146dca99f
Estimated completion time is ~0.9 hour(s)
centos 7.8.2003 (Linux 3.10.0-957.1.3.el7.x86_64)
 GCC: gcc (GCC) 7.3.1 20180303 (Red Hat 7.3.1-5)
 Host: ip-10-20-0-116.ca-central-1.compute.internal

HPCCSmoketest commented:
Automated Smoketest: :white_check_mark:
OS: centos 7.6.1810 (Linux 3.10.0-957.1.3.el7.x86_64)
 GCC: gcc (GCC) 7.3.1 20180303 (Red Hat 7.3.1-5)
 Host: ip-10-20-0-65.ca-central-1.compute.internal
Sha: f725998e723b13e5e20093ebe3a7dd7146dca99f
Build: success
Milestone:Install hpccsystems-platform-community_7.13.0-trunk0.el7.x86_64.rpm
HPCC Start: OK

Unit tests result:

| Test | total | passed | failed | errors | timeout | elaps | 
|---|---|---|---|---|---|---|
|unittest |145 |145 |0 |0 |0 |73 sec |
|wutoolTest(Dali) |19 |19 |0 |0 |0 |1 sec |
|wutoolTest(Cassandra) |19 |19 |0 |0 |0 |8 sec |



Regression test result:

| phase | total | pass | fail | elaps | 
|---|---|---|---|---|
|setup (hthor) |11 |11 |0 |30 sec  (00:00:30) |
|setup (thor) |11 |11 |0 |43 sec  (00:00:43) |
|setup (roxie) |11 |11 |0 |26 sec  (00:00:26) |
|test (hthor) |934 |934 |0 |376 sec  (00:06:16) |
|test (thor) |845 |845 |0 |883 sec  (00:14:43) |
|test (roxie) |1013 |1013 |0 |418 sec  (00:06:58) |


HPCC Stop: OK
HPCC Uninstall: OK
Time stats:

| Prep time | Build time | Package time | Install time | Start time | Test time | Stop time | Summary | 
|---|---|---|---|---|---|---|---|
|    9 sec (00:00:09) |  781 sec (00:13:01) |  121 sec (00:02:01) |   23 sec (00:00:23) |   19 sec (00:00:19) | 2028 sec (00:33:48) |   18 sec (00:00:18) | 2999 sec (00:49:59) |

HPCCSmoketest commented:
Automated Smoketest: :white_check_mark:
OS: centos 7.8.2003 (Linux 3.10.0-957.1.3.el7.x86_64)
 GCC: gcc (GCC) 7.3.1 20180303 (Red Hat 7.3.1-5)
 Host: ip-10-20-0-116.ca-central-1.compute.internal
Sha: f725998e723b13e5e20093ebe3a7dd7146dca99f
Build: success
Milestone:Install hpccsystems-platform-community_7.13.0-trunk0.el7.x86_64.rpm
HPCC Start: OK

Unit tests result:

| Test | total | passed | failed | errors | timeout | elaps | 
|---|---|---|---|---|---|---|
|unittest |145 |145 |0 |0 |0 |74 sec |
|wutoolTest(Dali) |19 |19 |0 |0 |0 |2 sec |
|wutoolTest(Cassandra) |19 |19 |0 |0 |0 |9 sec |



Regression test result:

| phase | total | pass | fail | elaps | 
|---|---|---|---|---|
|setup (hthor) |11 |11 |0 |29 sec  (00:00:29) |
|setup (thor) |11 |11 |0 |45 sec  (00:00:45) |
|setup (roxie) |11 |11 |0 |19 sec  (00:00:19) |
|test (hthor) |934 |934 |0 |672 sec  (00:11:12) |
|test (thor) |845 |845 |0 |1194 sec  (00:19:54) |
|test (roxie) |1013 |1013 |0 |774 sec  (00:12:54) |


HPCC Stop: OK
HPCC Uninstall: OK
Time stats:

| Prep time | Build time | Package time | Install time | Start time | Test time | Stop time | Summary | 
|---|---|---|---|---|---|---|---|
|    9 sec (00:00:09) | 1088 sec (00:18:08) |  126 sec (00:02:06) |   24 sec (00:00:24) |   19 sec (00:00:19) | 2989 sec (00:49:49) |   17 sec (00:00:17) | 4272 sec (01:11:12) |

ghalliday commented:
@richardkchapman I think this might be a reasonable candidate for a point release.  It changes very few examples, and those it does I think the changes are correct.  I have opened a jira related to the one change I spotted in the generated code.  It should probably be re-reviewed by shamser first though.

shamser reviewed (dismissed):
@richardkchapman Reviewed.

HPCCSmoketest commented:
Process of PR-13589, label: uniqueaction starts now.
The reason of this test is: Forced to re-test.
Commit ID: 11b0dd47d34a9bfe261f1ac81c34c69aabb67a34
Estimated completion time is ~0.8 hour(s)
centos 7.6.1810 (Linux 3.10.0-957.1.3.el7.x86_64)
 GCC: gcc (GCC) 7.3.1 20180303 (Red Hat 7.3.1-5)
 Host: ip-10-20-0-17.ca-central-1.compute.internal

HPCCSmoketest commented:
Automated Smoketest: :x:
OS: centos 7.6.1810 (Linux 3.10.0-957.1.3.el7.x86_64)
 GCC: gcc (GCC) 7.3.1 20180303 (Red Hat 7.3.1-5)
 Host: ip-10-20-0-17.ca-central-1.compute.internal
Sha: 11b0dd47d34a9bfe261f1ac81c34c69aabb67a34
Build: success
Milestone:Install hpccsystems-platform-community_7.13.0-trunk0.el7.x86_64.rpm
HPCC Start: OK

Unit tests result:

| Test | total | passed | failed | errors | timeout | elaps | 
|---|---|---|---|---|---|---|
|unittest |145 |145 |0 |0 |0 |73 sec |
|wutoolTest(Dali) |19 |19 |0 |0 |0 |2 sec |
|wutoolTest(Cassandra) |19 |19 |0 |0 |0 |7 sec |



Regression test result:

| phase | total | pass | fail | elaps | 
|---|---|---|---|---|
|setup (hthor) |11 |11 |0 |25 sec  (00:00:25) |
|setup (thor) |11 |11 |0 |43 sec  (00:00:43) |
|setup (roxie) |11 |11 |0 |24 sec  (00:00:24) |
|test (hthor) |934 |933 |1 |374 sec  (00:06:14) |
|test (thor) |845 |844 |1 |894 sec  (00:14:54) |
|test (roxie) |1013 |1012 |1 |425 sec  (00:07:05) |


[Errors:](https://gist.github.com/410a1d42817f20442df0210ad1321173)
- test
  - hthor
    - 794\. [teststdlibrary](https://gist.github.com/HPCCSmoketest/410a1d42817f20442df0210ad1321173#file-teststdlibrary-w20201020-162209-4-log), 
  - thor
    - 749\. [teststdlibrary](https://gist.github.com/HPCCSmoketest/410a1d42817f20442df0210ad1321173#file-teststdlibrary-w20201020-163252-5-log), 
  - roxie
    - 875\. [teststdlibrary](https://gist.github.com/HPCCSmoketest/410a1d42817f20442df0210ad1321173#file-teststdlibrary-w20201020-164438-1-log), 

HPCC Stop: OK
HPCC Uninstall: OK
Time stats:

| Prep time | Build time | Package time | Install time | Start time | Test time | Stop time | Summary | 
|---|---|---|---|---|---|---|---|
|    9 sec (00:00:09) |  769 sec (00:12:49) |  121 sec (00:02:01) |   23 sec (00:00:23) |   18 sec (00:00:18) | 2040 sec (00:34:00) |   19 sec (00:00:19) | 2999 sec (00:49:59) |

HPCCSmoketest commented:
Process of PR-13589, label: uniqueaction starts now.
The reason of this test is: New commit.
Commit ID: 11b0dd47d34a9bfe261f1ac81c34c69aabb67a34
Estimated completion time is ~0.9 hour(s)
centos 7.8.2003 (Linux 3.10.0-957.1.3.el7.x86_64)
 GCC: gcc (GCC) 7.3.1 20180303 (Red Hat 7.3.1-5)
 Host: ip-10-20-0-116.ca-central-1.compute.internal

HPCCSmoketest commented:
Automated Smoketest: :x:
OS: centos 7.8.2003 (Linux 3.10.0-957.1.3.el7.x86_64)
 GCC: gcc (GCC) 7.3.1 20180303 (Red Hat 7.3.1-5)
 Host: ip-10-20-0-116.ca-central-1.compute.internal
Sha: 11b0dd47d34a9bfe261f1ac81c34c69aabb67a34
Build: success
Milestone:Install hpccsystems-platform-community_7.13.0-trunk0.el7.x86_64.rpm
HPCC Start: OK

Unit tests result:

| Test | total | passed | failed | errors | timeout | elaps | 
|---|---|---|---|---|---|---|
|unittest |145 |145 |0 |0 |0 |73 sec |
|wutoolTest(Dali) |19 |19 |0 |0 |0 |1 sec |
|wutoolTest(Cassandra) |19 |19 |0 |0 |0 |9 sec |



Regression test result:

| phase | total | pass | fail | elaps | 
|---|---|---|---|---|
|setup (hthor) |11 |11 |0 |27 sec  (00:00:27) |
|setup (thor) |11 |11 |0 |45 sec  (00:00:45) |
|setup (roxie) |11 |11 |0 |19 sec  (00:00:19) |
|test (hthor) |934 |933 |1 |694 sec  (00:11:34) |
|test (thor) |845 |844 |1 |1134 sec  (00:18:54) |
|test (roxie) |1013 |1012 |1 |776 sec  (00:12:56) |


[Errors:](https://gist.github.com/426e32ebc54ec9282159a107b0bb9dc7)
- test
  - hthor
    - 794\. [teststdlibrary](https://gist.github.com/HPCCSmoketest/426e32ebc54ec9282159a107b0bb9dc7#file-teststdlibrary-w20201020-181248-log), 
  - thor
    - 749\. [teststdlibrary](https://gist.github.com/HPCCSmoketest/426e32ebc54ec9282159a107b0bb9dc7#file-teststdlibrary-w20201020-182846-8-log), 
  - roxie
    - 875\. [teststdlibrary](https://gist.github.com/HPCCSmoketest/426e32ebc54ec9282159a107b0bb9dc7#file-teststdlibrary-w20201020-184535-log), 

HPCC Stop: OK
HPCC Uninstall: OK
Time stats:

| Prep time | Build time | Package time | Install time | Start time | Test time | Stop time | Summary | 
|---|---|---|---|---|---|---|---|
|    9 sec (00:00:09) | 1084 sec (00:18:04) |  127 sec (00:02:07) |   24 sec (00:00:24) |   18 sec (00:00:18) | 2972 sec (00:49:32) |   18 sec (00:00:18) | 4252 sec (01:10:52) |

HPCCSmoketest commented:
Process of PR-13589, label: uniqueaction starts now.
The reason of this test is: New commit.
Commit ID: 72a5069a7ec7898509055bf9e9766e64fd3f9cf1
Estimated completion time is ~0.9 hour(s)
centos 7.8.2003 (Linux 3.10.0-957.1.3.el7.x86_64)
 GCC: gcc (GCC) 7.3.1 20180303 (Red Hat 7.3.1-5)
 Host: ip-10-20-0-116.ca-central-1.compute.internal

HPCCSmoketest commented:
Process of PR-13589, label: uniqueaction starts now.
The reason of this test is: Forced to re-test.
Commit ID: 72a5069a7ec7898509055bf9e9766e64fd3f9cf1
Estimated completion time is ~0.8 hour(s)
centos 7.6.1810 (Linux 3.10.0-957.1.3.el7.x86_64)
 GCC: gcc (GCC) 7.3.1 20180303 (Red Hat 7.3.1-5)
 Host: ip-10-20-0-184.ca-central-1.compute.internal

HPCCSmoketest commented:
Automated Smoketest: :x:
OS: centos 7.6.1810 (Linux 3.10.0-957.1.3.el7.x86_64)
 GCC: gcc (GCC) 7.3.1 20180303 (Red Hat 7.3.1-5)
 Host: ip-10-20-0-184.ca-central-1.compute.internal
Sha: 72a5069a7ec7898509055bf9e9766e64fd3f9cf1
Build: success
Milestone:Install hpccsystems-platform-community_7.13.0-trunk0.el7.x86_64.rpm
HPCC Start: OK

Unit tests result:

| Test | total | passed | failed | errors | timeout | elaps | 
|---|---|---|---|---|---|---|
|unittest |145 |145 |0 |0 |0 |72 sec |
|wutoolTest(Dali) |19 |19 |0 |0 |0 |1 sec |
|wutoolTest(Cassandra) |19 |19 |0 |0 |0 |8 sec |



Regression test result:

| phase | total | pass | fail | elaps | 
|---|---|---|---|---|
|setup (hthor) |11 |11 |0 |26 sec  (00:00:26) |
|setup (thor) |11 |11 |0 |43 sec  (00:00:43) |
|setup (roxie) |11 |11 |0 |24 sec  (00:00:24) |
|test (hthor) |934 |933 |1 |376 sec  (00:06:16) |
|test (thor) |845 |844 |1 |834 sec  (00:13:54) |
|test (roxie) |1013 |1009 |1 |N/A |


[Errors:](https://gist.github.com/308ab67590af49168664e7321d0452fa)
- test
  - hthor
    - 794\. [teststdlibrary](https://gist.github.com/HPCCSmoketest/308ab67590af49168664e7321d0452fa#file-teststdlibrary-w20201021-083806-2-log), 
  - thor
    - 749\. [teststdlibrary](https://gist.github.com/HPCCSmoketest/308ab67590af49168664e7321d0452fa#file-teststdlibrary-w20201021-084848-1-log), 
  - roxie
    - 875\. [teststdlibrary](https://gist.github.com/HPCCSmoketest/308ab67590af49168664e7321d0452fa#file-teststdlibrary-w20201021-085930-2-log), 

HPCC Stop: OK
HPCC Uninstall: OK

richardkchapman commented:
@ghalliday Can you explain the regression suite difference - and will any user queries be similarly affected ?

HPCCSmoketest commented:
Automated Smoketest: :x:
OS: centos 7.8.2003 (Linux 3.10.0-957.1.3.el7.x86_64)
 GCC: gcc (GCC) 7.3.1 20180303 (Red Hat 7.3.1-5)
 Host: ip-10-20-0-116.ca-central-1.compute.internal
Sha: 72a5069a7ec7898509055bf9e9766e64fd3f9cf1
Build: success
Milestone:Install hpccsystems-platform-community_7.13.0-trunk0.el7.x86_64.rpm
HPCC Start: OK

Unit tests result:

| Test | total | passed | failed | errors | timeout | elaps | 
|---|---|---|---|---|---|---|
|unittest |145 |145 |0 |0 |0 |75 sec |
|wutoolTest(Dali) |19 |19 |0 |0 |0 |1 sec |
|wutoolTest(Cassandra) |19 |19 |0 |0 |0 |9 sec |



Regression test result:

| phase | total | pass | fail | elaps | 
|---|---|---|---|---|
|setup (hthor) |11 |11 |0 |29 sec  (00:00:29) |
|setup (thor) |11 |11 |0 |45 sec  (00:00:45) |
|setup (roxie) |11 |11 |0 |19 sec  (00:00:19) |
|test (hthor) |934 |933 |1 |689 sec  (00:11:29) |
|test (thor) |845 |844 |1 |1194 sec  (00:19:54) |
|test (roxie) |1013 |1012 |1 |781 sec  (00:13:01) |


[Errors:](https://gist.github.com/28538e490e6fd2ddcb2923594fe1139f)
- test
  - hthor
    - 794\. [teststdlibrary](https://gist.github.com/HPCCSmoketest/28538e490e6fd2ddcb2923594fe1139f#file-teststdlibrary-w20201021-084655-log), 
  - thor
    - 749\. [teststdlibrary](https://gist.github.com/HPCCSmoketest/28538e490e6fd2ddcb2923594fe1139f#file-teststdlibrary-w20201021-090320-log), 
  - roxie
    - 875\. [teststdlibrary](https://gist.github.com/HPCCSmoketest/28538e490e6fd2ddcb2923594fe1139f#file-teststdlibrary-w20201021-092040-3-log), 

HPCC Stop: OK
HPCC Uninstall: OK
Time stats:

| Prep time | Build time | Package time | Install time | Start time | Test time | Stop time | Summary | 
|---|---|---|---|---|---|---|---|
|    9 sec (00:00:09) | 1066 sec (00:17:46) |  128 sec (00:02:08) |   23 sec (00:00:23) |   17 sec (00:00:17) | 3015 sec (00:50:15) |   17 sec (00:00:17) | 4275 sec (01:11:15) |

ghalliday commented:
It is because I realised we were running the slow part of the date regression tests twice - so I removed one call, but didn't correctly update the key file to match.

HPCCSmoketest commented:
Process of PR-13589, label: uniqueaction starts now.
The reason of this test is: New commit.
Commit ID: c9d9ae70d7c13e89f7131c72282a0f600f99f17b
Estimated completion time is ~0.9 hour(s)
centos 7.8.2003 (Linux 3.10.0-957.1.3.el7.x86_64)
 GCC: gcc (GCC) 7.3.1 20180303 (Red Hat 7.3.1-5)
 Host: ip-10-20-0-116.ca-central-1.compute.internal

HPCCSmoketest commented:
Process of PR-13589, label: uniqueaction starts now.
The reason of this test is: Forced to re-test.
Commit ID: c9d9ae70d7c13e89f7131c72282a0f600f99f17b
Estimated completion time is ~0.8 hour(s)
centos 7.6.1810 (Linux 3.10.0-957.1.3.el7.x86_64)
 GCC: gcc (GCC) 7.3.1 20180303 (Red Hat 7.3.1-5)
 Host: ip-10-20-0-219.ca-central-1.compute.internal

HPCCSmoketest commented:
Automated Smoketest: :white_check_mark:
OS: centos 7.6.1810 (Linux 3.10.0-957.1.3.el7.x86_64)
 GCC: gcc (GCC) 7.3.1 20180303 (Red Hat 7.3.1-5)
 Host: ip-10-20-0-219.ca-central-1.compute.internal
Sha: c9d9ae70d7c13e89f7131c72282a0f600f99f17b
Build: success
Milestone:Install hpccsystems-platform-community_7.13.0-trunk0.el7.x86_64.rpm
HPCC Start: OK

Unit tests result:

| Test | total | passed | failed | errors | timeout | elaps | 
|---|---|---|---|---|---|---|
|unittest |145 |145 |0 |0 |0 |73 sec |
|wutoolTest(Dali) |19 |19 |0 |0 |0 |2 sec |
|wutoolTest(Cassandra) |19 |19 |0 |0 |0 |7 sec |



Regression test result:

| phase | total | pass | fail | elaps | 
|---|---|---|---|---|
|setup (hthor) |11 |11 |0 |26 sec  (00:00:26) |
|setup (thor) |11 |11 |0 |44 sec  (00:00:44) |
|setup (roxie) |11 |11 |0 |21 sec  (00:00:21) |
|test (hthor) |934 |934 |0 |374 sec  (00:06:14) |
|test (thor) |845 |845 |0 |834 sec  (00:13:54) |
|test (roxie) |1013 |1013 |0 |470 sec  (00:07:50) |


HPCC Stop: OK
HPCC Uninstall: OK
Time stats:

| Prep time | Build time | Package time | Install time | Start time | Test time | Stop time | Summary | 
|---|---|---|---|---|---|---|---|
|    9 sec (00:00:09) |  772 sec (00:12:52) |  120 sec (00:02:00) |   24 sec (00:00:24) |   18 sec (00:00:18) | 2021 sec (00:33:41) |   19 sec (00:00:19) | 2983 sec (00:49:43) |

HPCCSmoketest commented:
Automated Smoketest: :white_check_mark:
OS: centos 7.8.2003 (Linux 3.10.0-957.1.3.el7.x86_64)
 GCC: gcc (GCC) 7.3.1 20180303 (Red Hat 7.3.1-5)
 Host: ip-10-20-0-116.ca-central-1.compute.internal
Sha: c9d9ae70d7c13e89f7131c72282a0f600f99f17b
Build: success
Milestone:Install hpccsystems-platform-community_7.13.0-trunk0.el7.x86_64.rpm
HPCC Start: OK

Unit tests result:

| Test | total | passed | failed | errors | timeout | elaps | 
|---|---|---|---|---|---|---|
|unittest |145 |145 |0 |0 |0 |77 sec |
|wutoolTest(Dali) |19 |19 |0 |0 |0 |2 sec |
|wutoolTest(Cassandra) |19 |19 |0 |0 |0 |8 sec |



Regression test result:

| phase | total | pass | fail | elaps | 
|---|---|---|---|---|
|setup (hthor) |11 |11 |0 |28 sec  (00:00:28) |
|setup (thor) |11 |11 |0 |44 sec  (00:00:44) |
|setup (roxie) |11 |11 |0 |20 sec  (00:00:20) |
|test (hthor) |934 |934 |0 |704 sec  (00:11:44) |
|test (thor) |845 |845 |0 |1134 sec  (00:18:54) |
|test (roxie) |1013 |1013 |0 |776 sec  (00:12:56) |


HPCC Stop: OK
HPCC Uninstall: OK
Time stats:

| Prep time | Build time | Package time | Install time | Start time | Test time | Stop time | Summary | 
|---|---|---|---|---|---|---|---|
|   10 sec (00:00:10) | 1060 sec (00:17:40) |  129 sec (00:02:09) |   24 sec (00:00:24) |   20 sec (00:00:20) | 2987 sec (00:49:47) |   17 sec (00:00:17) | 4247 sec (01:10:47) |

HPCCSmoketest commented:
Process of PR-13589, label: uniqueaction starts now.
The reason of this test is: Forced to re-test.
Commit ID: c9d9ae70d7c13e89f7131c72282a0f600f99f17b
Estimated completion time is ~0.75 hour(s)
centos 7.6.1810 (Linux 3.10.0-957.1.3.el7.x86_64)
 GCC: gcc (GCC) 7.3.1 20180303 (Red Hat 7.3.1-5)
 Host: ip-10-20-0-46.ca-central-1.compute.internal

HPCCSmoketest commented:
Automated Smoketest: :white_check_mark:
OS: centos 7.6.1810 (Linux 3.10.0-957.1.3.el7.x86_64)
 GCC: gcc (GCC) 7.3.1 20180303 (Red Hat 7.3.1-5)
 Host: ip-10-20-0-46.ca-central-1.compute.internal
Sha: c9d9ae70d7c13e89f7131c72282a0f600f99f17b
Build: success
Milestone:Install hpccsystems-platform-community_7.13.0-trunk0.el7.x86_64.rpm
HPCC Start: OK

Unit tests result:

| Test | total | passed | failed | errors | timeout | elaps | 
|---|---|---|---|---|---|---|
|unittest |145 |145 |0 |0 |0 |52 sec |
|wutoolTest(Dali) |19 |19 |0 |0 |0 |1 sec |
|wutoolTest(Cassandra) |19 |19 |0 |0 |0 |5 sec |



Regression test result:

| phase | total | pass | fail | elaps | 
|---|---|---|---|---|
|setup (hthor) |11 |11 |0 |24 sec  (00:00:24) |
|setup (thor) |11 |11 |0 |40 sec  (00:00:40) |
|setup (roxie) |11 |11 |0 |20 sec  (00:00:20) |
|test (hthor) |936 |936 |0 |307 sec  (00:05:07) |
|test (thor) |847 |847 |0 |715 sec  (00:11:55) |
|test (roxie) |1015 |1015 |0 |352 sec  (00:05:52) |


HPCC Stop: OK
HPCC Uninstall: OK
Time stats:

| Prep time | Build time | Package time | Install time | Start time | Test time | Stop time | Summary | 
|---|---|---|---|---|---|---|---|
|    9 sec (00:00:09) |  603 sec (00:10:03) |  107 sec (00:01:47) |   20 sec (00:00:20) |   16 sec (00:00:16) | 1693 sec (00:28:13) |   18 sec (00:00:18) | 2466 sec (00:41:06) |

HPCCSmoketest commented:
Process of PR-13589, label: uniqueaction starts now.
The reason of this test is: Forced to re-test.
Commit ID: c9d9ae70d7c13e89f7131c72282a0f600f99f17b
Estimated completion time is ~0.75 hour(s)
centos 7.6.1810 (Linux 3.10.0-957.1.3.el7.x86_64)
 GCC: gcc (GCC) 7.3.1 20180303 (Red Hat 7.3.1-5)
 Host: ip-10-20-0-41.ca-central-1.compute.internal

HPCCSmoketest commented:
Automated Smoketest: :white_check_mark:
OS: centos 7.6.1810 (Linux 3.10.0-957.1.3.el7.x86_64)
 GCC: gcc (GCC) 7.3.1 20180303 (Red Hat 7.3.1-5)
 Host: ip-10-20-0-41.ca-central-1.compute.internal
Sha: c9d9ae70d7c13e89f7131c72282a0f600f99f17b
Build: success
Milestone:Install hpccsystems-platform-community_7.13.0-trunk0.el7.x86_64.rpm
HPCC Start: OK

Unit tests result:

| Test | total | passed | failed | errors | timeout | elaps | 
|---|---|---|---|---|---|---|
|unittest |145 |145 |0 |0 |0 |54 sec |
|wutoolTest(Dali) |19 |19 |0 |0 |0 |1 sec |
|wutoolTest(Cassandra) |19 |19 |0 |0 |0 |5 sec |



Regression test result:

| phase | total | pass | fail | elaps | 
|---|---|---|---|---|
|setup (hthor) |9 |9 |0 |25 sec  (00:00:25) |
|setup (thor) |9 |9 |0 |40 sec  (00:00:40) |
|setup (roxie) |19 |19 |0 |18 sec  (00:00:18) |
|test (hthor) |947 |947 |0 |290 sec  (00:04:50) |
|test (thor) |858 |858 |0 |776 sec  (00:12:56) |
|test (roxie) |1020 |1020 |0 |351 sec  (00:05:51) |


HPCC Stop: OK
HPCC Uninstall: OK
Time stats:

| Prep time | Build time | Package time | Install time | Start time | Test time | Stop time | Summary | 
|---|---|---|---|---|---|---|---|
|    9 sec (00:00:09) |  604 sec (00:10:04) |  113 sec (00:01:53) |   20 sec (00:00:20) |   16 sec (00:00:16) | 1737 sec (00:28:57) |   17 sec (00:00:17) | 2516 sec (00:41:56) |

HPCCSmoketest commented:
Process of PR-13589, label: uniqueaction starts now.
The reason of this test is: Forced to re-test.
Commit ID: c9d9ae70d7c13e89f7131c72282a0f600f99f17b
Estimated completion time is ~0.75 hour(s)
centos 7.6.1810 (Linux 3.10.0-957.1.3.el7.x86_64)
 GCC: gcc (GCC) 7.3.1 20180303 (Red Hat 7.3.1-5)
 Host: ip-10-20-0-10.ca-central-1.compute.internal

HPCCSmoketest commented:
Automated Smoketest: :white_check_mark:
OS: centos 7.6.1810 (Linux 3.10.0-957.1.3.el7.x86_64)
 GCC: gcc (GCC) 7.3.1 20180303 (Red Hat 7.3.1-5)
 Host: ip-10-20-0-10.ca-central-1.compute.internal
Sha: c9d9ae70d7c13e89f7131c72282a0f600f99f17b
Build: success
Milestone:Install hpccsystems-platform-community_7.13.0-trunk0.el7.x86_64.rpm
HPCC Start: OK

Unit tests result:

| Test | total | passed | failed | errors | timeout | elaps | 
|---|---|---|---|---|---|---|
|unittest |145 |145 |0 |0 |0 |52 sec |
|wutoolTest(Dali) |19 |19 |0 |0 |0 |1 sec |
|wutoolTest(Cassandra) |19 |19 |0 |0 |0 |5 sec |



Regression test result:

| phase | total | pass | fail | elaps | 
|---|---|---|---|---|
|setup (hthor) |9 |9 |0 |23 sec  (00:00:23) |
|setup (thor) |9 |9 |0 |40 sec  (00:00:40) |
|setup (roxie) |19 |19 |0 |20 sec  (00:00:20) |
|test (hthor) |947 |947 |0 |298 sec  (00:04:58) |
|test (thor) |858 |858 |0 |772 sec  (00:12:52) |
|test (roxie) |1020 |1020 |0 |352 sec  (00:05:52) |


HPCC Stop: OK
HPCC Uninstall: OK
Time stats:

| Prep time | Build time | Package time | Install time | Start time | Test time | Stop time | Summary | 
|---|---|---|---|---|---|---|---|
|    9 sec (00:00:09) |  605 sec (00:10:05) |  108 sec (00:01:48) |   24 sec (00:00:24) |   16 sec (00:00:16) | 1742 sec (00:29:02) |   18 sec (00:00:18) | 2522 sec (00:42:02) |

HPCCSmoketest commented:
Process of PR-13589, label: uniqueaction starts now.
The reason of this test is: Forced to re-test.
Commit ID: c9d9ae70d7c13e89f7131c72282a0f600f99f17b
Estimated completion time is ~0.75 hour(s)
centos 7.6.1810 (Linux 3.10.0-957.1.3.el7.x86_64)
 GCC: gcc (GCC) 7.3.1 20180303 (Red Hat 7.3.1-5)
 Host: ip-10-20-0-134.ca-central-1.compute.internal

HPCCSmoketest commented:
Automated Smoketest: :white_check_mark:
OS: centos 7.6.1810 (Linux 3.10.0-957.1.3.el7.x86_64)
 GCC: gcc (GCC) 7.3.1 20180303 (Red Hat 7.3.1-5)
 Host: ip-10-20-0-134.ca-central-1.compute.internal
Sha: c9d9ae70d7c13e89f7131c72282a0f600f99f17b
Build: success
Milestone:Install hpccsystems-platform-community_7.13.0-trunk0.el7.x86_64.rpm
HPCC Start: OK

Unit tests result:

| Test | total | passed | failed | errors | timeout | elaps | 
|---|---|---|---|---|---|---|
|unittest |145 |145 |0 |0 |0 |51 sec |
|wutoolTest(Dali) |19 |19 |0 |0 |0 |1 sec |
|wutoolTest(Cassandra) |19 |19 |0 |0 |0 |5 sec |



Regression test result:

| phase | total | pass | fail | elaps | 
|---|---|---|---|---|
|setup (hthor) |9 |9 |0 |21 sec  (00:00:21) |
|setup (thor) |9 |9 |0 |40 sec  (00:00:40) |
|setup (roxie) |19 |19 |0 |21 sec  (00:00:21) |
|test (hthor) |947 |947 |0 |309 sec  (00:05:09) |
|test (thor) |858 |858 |0 |775 sec  (00:12:55) |
|test (roxie) |1020 |1020 |0 |353 sec  (00:05:53) |


HPCC Stop: OK
HPCC Uninstall: OK
Time stats:

| Prep time | Build time | Package time | Install time | Start time | Test time | Stop time | Summary | 
|---|---|---|---|---|---|---|---|
|    8 sec (00:00:08) |  598 sec (00:09:58) |  105 sec (00:01:45) |   20 sec (00:00:20) |   17 sec (00:00:17) | 1754 sec (00:29:14) |   17 sec (00:00:17) | 2519 sec (00:41:59) |

ghalliday commented:
@richardkchapman I have squashed the previous code and fixed a memory leak.  The vast majority of the changes are coping with attributes on things like IF() actions and SEQUENTIAL.  The final form of the PR doesn't actually add them, so they could theoretically be removed - but it is probably better if they are kept in case we ever want to support attributes everywhere (e.g., like HINT)

HPCCSmoketest commented:
Process of PR-13589, label: uniqueaction starts now.
The reason of this test is: Forced to re-test.
Commit ID: f18da485e4a0c18a0b0c8e5d9ebbd1c2c231459f
Estimated completion time is ~0.75 hour(s)
centos 7.6.1810 (Linux 3.10.0-957.1.3.el7.x86_64)
 GCC: gcc (GCC) 7.3.1 20180303 (Red Hat 7.3.1-5)
 Host: ip-10-20-0-147.ca-central-1.compute.internal

HPCCSmoketest commented:
Automated Smoketest: :white_check_mark:
OS: centos 7.6.1810 (Linux 3.10.0-957.1.3.el7.x86_64)
 GCC: gcc (GCC) 7.3.1 20180303 (Red Hat 7.3.1-5)
 Host: ip-10-20-0-147.ca-central-1.compute.internal
Sha: f18da485e4a0c18a0b0c8e5d9ebbd1c2c231459f
Build: success
Milestone:Install hpccsystems-platform-community_7.13.0-trunk0.el7.x86_64.rpm
HPCC Start: OK

Unit tests result:

| Test | total | passed | failed | errors | timeout | elaps | 
|---|---|---|---|---|---|---|
|unittest |145 |145 |0 |0 |0 |52 sec |
|wutoolTest(Dali) |19 |19 |0 |0 |0 |1 sec |
|wutoolTest(Cassandra) |19 |19 |0 |0 |0 |5 sec |



Regression test result:

| phase | total | pass | fail | elaps | 
|---|---|---|---|---|
|setup (hthor) |9 |9 |0 |25 sec  (00:00:25) |
|setup (thor) |9 |9 |0 |41 sec  (00:00:41) |
|setup (roxie) |19 |19 |0 |17 sec  (00:00:17) |
|test (hthor) |947 |947 |0 |669 sec  (00:11:09) |
|test (thor) |858 |858 |0 |1130 sec  (00:18:50) |
|test (roxie) |1020 |1020 |0 |722 sec  (00:12:02) |


HPCC Stop: OK
HPCC Uninstall: OK
Time stats:

| Prep time | Build time | Package time | Install time | Start time | Test time | Stop time | Summary | 
|---|---|---|---|---|---|---|---|
|    8 sec (00:00:08) |  606 sec (00:10:06) |  109 sec (00:01:49) |   20 sec (00:00:20) |   20 sec (00:00:20) | 1435 sec (00:23:55) |   19 sec (00:00:19) | 2217 sec (00:36:57) |
